### PR TITLE
feat(GCP): Exfiltrates a Compute Image by sharing it

### PR DESCRIPTION
**Pull request by vthiery**
### What does this PR do?

Introduces `gcp.exfiltration.share-compute-image` where data is exfiltrated by sharing a Compute Image.

### Motivation

Inspired by [https://www.mitiga.io/blog/google-cloud-platform-exfiltration-a-threat-hunting-guide mentionned in ](https://github.com/DataDog/stratus-red-team/issues/371)

### To answer before getting out of draft

- [ ] Can we make it not "Slow"?

### CODE REVIEW

File: v2/internal/attacktechniques/gcp/exfiltration/share-compute-image/main.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+          }
+        ]
+      },
+      "@type": "type.googleapis.com/compute.images.setIamPolicy"
+    }
+  }
+}
+` + codeBlock + `
+
+After the attacker has permissions on the Compute Image, they can export it in their own GCP Storage using:
```
Comment (christophetd) [inline v2/internal/attacktechniques/gcp/exfiltration/share-compute-image/main.go:84]: does the attacker need permissions on the image before doing this? wouldn't it be simpler to not share it, and directly export it to an attacker-controlled bucket?

File: v2/internal/attacktechniques/gcp/exfiltration/share-compute-image/main.tf [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+locals {
+  image-name = "stratus-red-team-victim-image"
+}
+
+resource "google_compute_image" "this" {
+  name = local.image-name
+
+  raw_disk {
+    source = "https://storage.googleapis.com/bosh-gce-raw-stemcells/bosh-stemcell-97.98-google-kvm-ubuntu-xenial-go_agent-raw-1557960142.tar.gz"
```
Comment (christophetd) [inline v2/internal/attacktechniques/gcp/exfiltration/share-compute-image/main.tf:18]: where is this coming from?

**Comment by christophetd** [comment]
Thanks for the PR! Looks great overall, left a few comments

File: v2/internal/attacktechniques/gcp/exfiltration/share-compute-image/main.tf (hunk 2 above)
Comment (vthiery) [reply v2/internal/attacktechniques/gcp/exfiltration/share-compute-image/main.tf:18]: I copied it from https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_image 
This is something I still need to look into in order to make it faster

Comment (christophetd) [reply v2/internal/attacktechniques/gcp/exfiltration/share-compute-image/main.tf:18]: I don't have a better way, so I'd suggest we merge as it - good for you?

File: v2/internal/attacktechniques/gcp/exfiltration/share-compute-image/main.go (hunk 1 above)
Comment (vthiery) [reply v2/internal/attacktechniques/gcp/exfiltration/share-compute-image/main.go:84]: From what I understood, sharing the image relies on the fact that the attacker obtain permissions to use `compute.images.setIamPolicy`. After doing so, the attacker can freely access the image from her own project.

On the other hand, exporting to an attacker-controlled bucket is much more involved as it requires using the Cloud Build API, and therefore require to either have permissions to enable it, or permissions to use it. Also, it is likely to be much noisier. 

Nevertheless, the latter would make an interesting use-case to add to `stratus-red-team` too.

**Review by christophetd — APPROVED** [review]

**Comment by christophetd** [comment]
will be released as part of v2.12.0 shortly

**Pull request merged**